lib/internal/util/comparisons.js

```diff
@@ -170,6 +179,12 @@ function looseDeepEqual(val1, val2, memos) {
   } else if (isArguments(val1Tag) || isArguments(val2Tag)) {
     return false;
   }
+  if (isArrayBuffer(val1Tag) && isArrayBuffer(val2Tag)) {
+    if (!areSimilarTypedArrays(new Uint8Array(val1),
+                               new Uint8Array(val2), 300)) {
```

Comment:
I would disagree that `assert.deepEqual(a, b)` should pass. Certainly `assert.deepEqual(new Float32Array(a), new Float32Array(b))` should pass, since the buffers contain loosely equal values when interpreted as float arrays. However, the underlying buffers do contain different bytes, so there is no reason to consider them equal without knowing that they are being used as float arrays
````javascript
// On a LE machine:
new Uint8Array(a) // Uint8Array(4) [0, 0, 0, 128]
new Uint8Array(b) // Uint8Array(4) [0, 0, 0, 0]
````

Also, changing to `Infinity` wouldn't affect that; it simply changes whether the bytes are compared by a JS `for` loop or by calling into the `Buffer.compare` C++ code.

Reply:
Hm, true, my bad.